src/storage/adapter/postgres.rs

```diff
@@ -0,0 +1,58 @@
+mod dek;
+
+use error_stack::ResultExt;
+
+use crate::storage::{adapter::PostgreSQL, errors, Config, Connection, DbState};
+use diesel_async::pooled_connection::{bb8::Pool, AsyncDieselConnectionManager, ManagerConfig};
+use diesel_async::AsyncPgConnection;
+use masking::PeekInterface;
+
+#[async_trait::async_trait]
+impl super::DbAdapter for DbState<PostgreSQL> {
+    type Conn<'a> = Connection<'a>;
+    type AdapterType = PostgreSQL;
+
+    /// # Panics
+    ///
+    /// Panics if unable to connect to Database
+    #[allow(clippy::expect_used)]
+    async fn from_config(config: &Config) -> Self {
+        let database = &config.database;
+
+        let password = database.password.expose(config).await;
+
+        let database_url = format!(
+            "postgres://{}:{}@{}:{}/{}",
+            database.user.peek(),
+            password.peek(),
+            database.host,
+            database.port,
+            database.dbname.peek()
+        );
+
+        let mgr_config = ManagerConfig::default();
+        let mgr = AsyncDieselConnectionManager::<AsyncPgConnection>::new_with_config(
+            database_url,
+            mgr_config,
+        );
+        let pool = Pool::builder()
+            .max_size(database.pool_size.unwrap_or(10))
```

Comment:
Can we preferably read this from config?

Reply:
This is read from config right? If it's not given the value will be 10